src/arithmetic/algebraic_expression.c

```diff
@@ -264,10 +245,24 @@ static QGNode **_DeepestLevel(const QueryGraph *g, int *level) {
 	return leafs;
 }
 
+/* If the edge is referenced or of a variable length, it should populate the AlgebraicExpression. */
+static inline bool _should_populate_edge(QGEdge *e) {
+	return (_referred_entity(e->alias) || QGEdge_VariableLength(e));
+}
+
+static inline bool _should_divide_expression(AlgebraicExpression *exp, QGEdge **path, int idx) {
+	QGEdge *e = path[idx];
+
+	return (_should_populate_edge(e)                  ||      // This edge is populated.
+			_should_populate_edge(path[idx + 1])      ||      // The next edge is populated.
```

Comment:
Not quite, actually! That for loop doesn't check the last edge just because the last edge by definition can't be an intermediate expression.

This is actually a way of handling patterns like `(a)-[]->(b)-[e]->() RETURN e`, which needs to generate the plan:
```
1) "Results"
2) "    Project"
3) "        Conditional Traverse | (b)-[e]->(anon_1)"
4) "            Conditional Traverse | (a)->(b)"
5) "                All Node Scan | (a)"
```

But without lookahead, we'll only create one pattern and one traverse op. On master, this is resolved by defining `b` as a referenced entity (a kludge solution). My alternate kludge solution here is lookahead. Any thoughts on alternatives?